irods_testing_environment/database_setup.py

```diff
@@ -231,6 +232,11 @@ def connect_to_database(self,
         name -- name of the database to check
         as_user -- name of the user/role connecting to the database
         """
+
+        # Make sure the database is ready for connections.
+        while self.execute_mysql_command('SHOW DATABASES;') != 0:
+            time.sleep(5)
```

Comment:
5 seconds seems like a lot to me, but if it seems appropriate based on your experience, I'm fine with it.

Also, consider adding a call to `logging.debug` so we can track what's hanging, should there be hangs.